libs/vscode/tasks/src/lib/nx-task-commands.ts

```diff
@@ -341,15 +354,58 @@ async function promptForList() {
 
 async function promptForMigrate() {
   const telemetry = getTelemetry();
-  telemetry.featureUsed('affected-cli');
-  const task = NxTask.create(
-    {
-      command: 'migrate',
-      flags: [],
-    },
-    cliTaskProvider.getWorkspacePath()
+  telemetry.featureUsed('migrate command');
+
+  const { workspacePath } = await getNxWorkspace();
+
+  const packageJson = await readAndParseJson(
+    join(workspacePath, 'package.json')
   );
-  tasks.executeTask(task);
+
+  if (
+    packageJson.dependencies == undefined ||
+    packageJson.devDependencies == undefined
+  ) {
+    return;
+  }
+
+  const dependencyToMigrate = await new Promise<string | undefined>(
+    (resolve) => {
+      const quickPick = window.createQuickPick();
+      quickPick.title = 'Select dependency';
+      quickPick.items = buildQuickPickItems(packageJson);
+      quickPick.placeholder = 'Select the dependency you want to migrate.';
+
+      quickPick.onDidAccept(() => {
+        resolve(quickPick.selectedItems[0]?.label);
+        quickPick.hide();
+      });
+
+      quickPick.show();
+    }
+  );
+
+  if (dependencyToMigrate === undefined) {
+    return;
+  }
+
+  const depVersioningInfo = await resolveDependencyVersioning(
+    dependencyToMigrate
+  );
+
```

Comment:
This was the case before you but how about a quick check here or in `dependency-versioning.ts` about whether the dep we're trying to install exists? 
Because if we type in garbage in the dependency quickpick right now, nothing happens. Probably a simple error notification 'The dependency ${dependency} you've selected doesn't exist' or something.

Reply:
After doing some quick tests, it seems that `showQuickPick` either returns `QuickPickItem` whenever an item from the list is selected, or `undefined` whenever the list is dismissed without selecting anything. It's not possible to submit random text. It must be an item from the list. I've added a notification in case the list is dismissed.

As for selecting a version, dismissing the list doesn't return anything from `resolveDependencyVersioning`. So `promptForVersion` should probably be tweaked to account for that case.

Reply:
Ah perfect! With the earlier implementation it was still possible but using `showQuickPick` fixed it!

I've removed the info message on cancel because it doesn't fit with our other flows when we cancel quickpicks 
:)